src/fetchtastic/download/orchestrator.py

```
@@ -217,6 +227,94 @@ def run_download_pipeline(
+            ]
+            tracked_tag = self.firmware_downloader.get_latest_release_tag()
+            new_versions: List[str] = []
+            for release in stable_releases:
+                if (
+                    tracked_tag is None
+                    or self.version_manager.compare_versions(
+                        release.tag_name, tracked_tag
+                    )
+                    > 0
+                ):
+                    new_versions.append(release.tag_name)
```

Ruff (0.15.10)

[warning] Use a list comprehension to create a transformed list

(PERF401)